fix: refresh git-backed WordPress extension after upstream fixes

## Problem

Data Machine PR #1571 still failed `Homeboy (lint)` and `Homeboy (test)` after rerunning failed checks, even though the exact WordPress extension routing bugs were already fixed upstream:

- ESLint over scoped PHP files: Extra-Chill/homeboy-extensions#325, fixed by Extra-Chill/homeboy-extensions#341 and #343.
- Playground test backend ignoring changed-test scope / schema setup: Extra-Chill/homeboy-extensions#327, fixed by Extra-Chill/homeboy-extensions#330.

PR run rerun after those fixes:

- PR: https://github.com/Extra-Chill/data-machine/pull/1571
- Run: https://github.com/Extra-Chill/data-machine/actions/runs/25086050739
- Rerun lint job: https://github.com/Extra-Chill/data-machine/actions/runs/25086050739/job/73502981890
- Rerun test job: https://github.com/Extra-Chill/data-machine/actions/runs/25086050739/job/73502981858

The rerun environment reports:

```text
HOMEBOY_CLI_VERSION: homeboy 0.123.1+b18962a4
HOMEBOY_EXTENSION_ID: wordpress
HOMEBOY_EXTENSION_SOURCE: https://github.com/Extra-Chill/homeboy-extensions
HOMEBOY_EXTENSION_REVISION: unknown
HOMEBOY_ACTION_REF: v2
```

## Evidence

### Lint still runs ESLint over PHP files

The rerun scoped lint to five changed PHP files:

```text
Linting 5 files matching: {/home/runner/work/data-machine/data-machine/inc/Core/Steps/AI/AIStep.php,/home/runner/work/data-machine/data-machine/inc/Core/Steps/AI/ToolPolicy/PipelineToolPolicyArgs.php,/home/runner/work/data-machine/data-machine/inc/Engine/AI/RequestInspector.php,/home/runner/work/data-machine/data-machine/inc/Engine/AI/Tools/ToolPolicyResolver.php,/home/runner/work/data-machine/data-machine/tests/pipeline-tool-policy-snapshot-smoke.php}
PHPCS linting passed
```

Then ESLint parsed those same PHP files:

```text
/home/runner/work/data-machine/data-machine/inc/Core/Steps/AI/AIStep.php
  1:1  error  Parsing error: Unexpected token (1:1)

/home/runner/work/data-machine/data-machine/inc/Core/Steps/AI/ToolPolicy/PipelineToolPolicyArgs.php
  1:1  error  Parsing error: Unexpected token (1:1)
```

That is the behavior #325/#341/#343 were meant to eliminate.

### Test still runs the full PHPUnit suite despite scoped file selection

The rerun output says Homeboy selected one changed test file:

```json
"test_scope": {
  "changed_since": "67bb09cfd426731edf85c5841cf903bd364d02f8",
  "mode": "changed",
  "selected_count": 1,
  "selected_files": [
    "tests/pipeline-tool-policy-snapshot-smoke.php"
  ]
}
```

But the runner executed the full PHPUnit suite:

```text
Tests: 1209, Assertions: 2736, Errors: 197, Failures: 174, Skipped: 4, Risky: 1.
```

That is the behavior #327/#330 were meant to eliminate.

## Expected behavior

- A rerun after the upstream extension fixes should use a WordPress extension revision that contains #330/#341/#343.
- The CI log should expose the actual extension revision instead of `HOMEBOY_EXTENSION_REVISION: unknown`.
- Extension caching/install should not keep using stale runner code after fixes merge upstream.

## Acceptance criteria

- Homeboy Action setup refreshes installed git-backed extensions when the source is `https://github.com/Extra-Chill/homeboy-extensions` and no explicit revision is pinned.
- CI logs show a real `HOMEBOY_EXTENSION_REVISION` SHA for git-backed extensions.
- Re-running Extra-Chill/data-machine#1571-style scoped lint/test jobs after extension fixes no longer uses stale runner behavior.

## AI assistance

- **AI assistance:** Yes
- **Tool(s):** OpenCode (GPT-5.5)
- **Used for:** Diagnosed Data Machine PR #1571 CI rerun logs, linked them to already-merged Homeboy extension fixes, and drafted this issue for Chris to review.

--- a/scripts/setup/capture-tooling-metadata.sh
+++ b/scripts/setup/capture-tooling-metadata.sh
@@ -30,6 +30,11 @@ if [ -n "${EXTENSION_ID_EFFECTIVE}" ] && [ "${EXTENSION_ID_EFFECTIVE}" != "auto"
   EXT_DIR="${HOME}/.config/homeboy/extensions/${EXTENSION_ID_EFFECTIVE}"
   if [ -d "${EXT_DIR}/.git" ]; then
     EXTENSION_REVISION="$(git -C "${EXT_DIR}" rev-parse --short HEAD 2>/dev/null || echo 'unknown')"
+  elif [ -f "${EXT_DIR}/.source-revision" ]; then
+    EXTENSION_REVISION="$(tr -d '[:space:]' < "${EXT_DIR}/.source-revision")"
+    if [ -z "${EXTENSION_REVISION}" ]; then
+      EXTENSION_REVISION="unknown"
+    fi
   fi
 fi
 

--- a/scripts/setup/install-extension.sh
+++ b/scripts/setup/install-extension.sh
@@ -23,6 +23,19 @@ refresh_extension() {
   homeboy extension uninstall "${extension_id}" >/dev/null 2>&1 || true
 }
 
+refresh_configured_extensions() {
+  local config_file="${COMPONENT_DIR}/homeboy.json"
+
+  if [ ! -f "${config_file}" ]; then
+    refresh_extension "${EXTENSION_ID}"
+    return 0
+  fi
+
+  while IFS= read -r configured_extension_id; do
+    refresh_extension "${configured_extension_id}"
+  done < <(jq -r '.extensions // {} | keys[]' "${config_file}" 2>/dev/null || true)
+}
+
 if [ -n "${EXTENSION_INPUT}" ]; then
   echo "Installing extension override: ${EXTENSION_INPUT} from ${EXTENSION_SOURCE}..."
   refresh_extension "${EXTENSION_INPUT}"
@@ -31,7 +44,7 @@ if [ -n "${EXTENSION_INPUT}" ]; then
 else
   if homeboy extension install-for-component --help >/dev/null 2>&1; then
     echo "Installing extensions configured by ${COMPONENT_DIR}/homeboy.json from ${EXTENSION_SOURCE}..."
-    refresh_extension "${EXTENSION_ID}"
+    refresh_configured_extensions
     homeboy extension install-for-component --path "${COMPONENT_DIR}" --source "${EXTENSION_SOURCE}"
     echo "Configured extensions installed successfully"
   else

--- a/scripts/setup/test-capture-tooling-metadata.sh
+++ b/scripts/setup/test-capture-tooling-metadata.sh
@@ -1,0 +1,48 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+ROOT_DIR="$(cd "${SCRIPT_DIR}/../.." && pwd)"
+
+assert_contains() {
+  local needle="$1"
+  local haystack="$2"
+  local label="$3"
+
+  if ! grep -Fq -- "${needle}" "${haystack}"; then
+    printf 'FAIL: %s\nmissing: %s\n' "${label}" "${needle}"
+    exit 1
+  fi
+
+  printf 'PASS: %s\n' "${label}"
+}
+
+TMP_DIR="$(mktemp -d)"
+trap 'rm -rf "${TMP_DIR}"' EXIT
+
+mkdir -p "${TMP_DIR}/bin" "${TMP_DIR}/home/.config/homeboy/extensions/wordpress"
+cat > "${TMP_DIR}/bin/homeboy" <<'STUB'
+#!/usr/bin/env bash
+if [ "$1" = "--version" ]; then
+  printf 'homeboy 9.9.9\n'
+fi
+STUB
+chmod +x "${TMP_DIR}/bin/homeboy"
+
+printf 'abc1234\n' > "${TMP_DIR}/home/.config/homeboy/extensions/wordpress/.source-revision"
+
+export PATH="${TMP_DIR}/bin:${PATH}"
+export HOME="${TMP_DIR}/home"
+export PORTABLE_EXTENSION="wordpress"
+export EXTENSION_SOURCE="https://github.com/Extra-Chill/homeboy-extensions"
+export ACTION_REF="v2"
+export ACTION_REPOSITORY="Extra-Chill/homeboy-action"
+export GITHUB_ENV="${TMP_DIR}/github-env"
+
+bash "${ROOT_DIR}/scripts/setup/capture-tooling-metadata.sh" > "${TMP_DIR}/metadata.log"
+
+assert_contains "HOMEBOY_EXTENSION_REVISION=abc1234" "${GITHUB_ENV}" "source revision file is exported"
+assert_contains "- Extension revision: abc1234" "${TMP_DIR}/metadata.log" "source revision file is logged"
+
+echo "All tooling metadata checks passed."

--- a/scripts/setup/test-install-extension.sh
+++ b/scripts/setup/test-install-extension.sh
@@ -34,14 +34,25 @@ chmod +x "${TMP_DIR}/bin/homeboy"
 export PATH="${TMP_DIR}/bin:${PATH}"
 export EXTENSION_SOURCE="/extensions"
 export EXTENSION_ID="wordpress"
-export COMPONENT_DIR="packages/plugin"
+export COMPONENT_DIR="${TMP_DIR}/component"
 export HOMEBOY_CALL_LOG="${TMP_DIR}/calls.log"
+
+mkdir -p "${COMPONENT_DIR}"
+cat > "${COMPONENT_DIR}/homeboy.json" <<'JSON'
+{
+  "id": "demo",
+  "extensions": {
+    "wordpress": {},
+    "nodejs": {}
+  }
+}
+JSON
 
 EXTENSION_INPUT="" bash "${ROOT_DIR}/scripts/setup/install-extension.sh" >/dev/null
 assert_equals \
-  $'extension uninstall wordpress\nextension install-for-component --path packages/plugin --source /extensions' \
+  $'extension uninstall nodejs\nextension uninstall wordpress\nextension install-for-component --path '"${COMPONENT_DIR}"$' --source /extensions' \
   "$(cat "${HOMEBOY_CALL_LOG}")" \
-  "configured extensions refresh cached copy before install-for-component"
+  "all configured extensions refresh cached copy before install-for-component"
 
 : > "${HOMEBOY_CALL_LOG}"
 HOMEBOY_INSTALL_FOR_COMPONENT_HELP_STATUS="1" EXTENSION_INPUT="" bash "${ROOT_DIR}/scripts/setup/install-extension.sh" >/dev/null

